src/qutip_tensornetwork/core/data/tensor_train/tensor_train.py

```diff
@@ -0,0 +1,289 @@
+        raise ValueError(
+            f" the bond shape between the last and the previoust"
+            " to last node is "
+            f" different ({previous_lbond_dim} and"
+            f" {nodes[-1].shape[-1]} respectively)."
+        )
+
+
+def network_to_tt(network, copy=True):
+    """This function transforms an arbitrary network into a tensor train. This
+    is done by first contracting the whole network into a single tensor an
+    then splitting it into a tensor-train by repeatedly applying an svd
+    transformation to the nodes. No truncation is done by this function. Hence
+    the output tensor-train represents exactly the input network.
+
+    For a more detailed explanation of the algorithm see [1]_.
+
+    Parameters
+    ----------
+    network : Network
+
+    copy: bool
+        If False, the operation is performed in-place.
+
+    References
+    ----------
+    .. [1] Paeckel, S., Köhler, T., Swoboda, A., Manmana, S. R., Schollwöck,
+    U., & Hubig, C. (2019). Time-evolution methods for matrix-product states.
+    Annals of Physics, 411, 167998.
+    """
+    if network.dims[0] != network.dims[1] and network.in_edges and network.out_edges:
+        raise NotImplementedError(
+            " At this moment the FiniteTT class can"
+            " only represent square matrices, kets"
+            " and bras."
+        )
+    network = network.contract(copy=copy)
+
+    n_nodes = max(len(network.in_edges), len(network.out_edges))
+    if network.in_edges:
+        in_edges = [[e] for e in network.in_edges]
+    else:
+        in_edges = [[] * n_nodes]
+
+    if network.out_edges:
+        out_edges = [[e] for e in network.in_edges]
+    else:
+        out_edges = [[] * n_nodes]
+
+    axes_names = ["out"] if network.out_edges else []
+    axes_names += ["in"] if network.in_edges else []
+
+    nodes = []
+    lbond = []
+    for i in range(n_nodes - 1):
+        left_edges = [network.out_edges[i]] if network.out_edges else []
+        left_edges += [network.in_edges[i]] if network.in_edges else []
+        left_edges += lbond
+
+        right_edges = network.out_edges[i + 1 :] if network.out_edges else []
+        right_edges += network.in_edges[i + 1 :] if network.in_edges else []
+
+        node = left_edges[0].node1
+
+        lnode, rnode, _ = tn.split_node(node, left_edges, right_edges)
+
+        rbond = [rnode[0]]
+        lnode.name = f"node_{i}"
+        lnode.reorder_edges(left_edges + rbond)
+        lnode.add_axis_names(axes_names + ["lbond"] * len(lbond) + ["rbond"])
+        nodes.append(lnode)
+
+        lbond = rbond
+
+    # For a single node we do not go through the for loop so we accomodate the
+    # variables here
+    if n_nodes == 1:
+        right_edges = [network.out_edges[0]] if network.out_edges else []
+        right_edges += [network.in_edges[0]] if network.in_edges else []
+        node = right_edges[0].node1
+        node.name = f"node_{n_nodes-1}"
```

Comment:
```suggestion
        node.name = f"node_0"
```
I think we should just write `node_0` here since this we already referred to this as the first node in the two lines above (and not the last node, even though the two ways of describing it are equivalent since there is only one node).